File: PO/Project/JavaProject/src/Plants/Grass.java
package Plants;

public class Grass extends Plant {

    public Grass(Point position, World world) {
        super(0, position, world);
        this.avatar = 'g';
    }

    @Override
    public Organism makeChild(Point newPosition) {
        return new Grass(newPosition, world);
    }

    @Override
    protected void finalize() throws Throwable {
        super.finalize();
        // Custom cleanup code, if necessary
    }
}

File: Sem2/PO/Project/C++/src/grass.cpp
#include "grass.h"

grass::grass(YX position, world* world_point)
: plant(0, position ,world_point)
{
    avatar = 'g';
}

organizm* grass::MakeChild(YX new_position)
{
    return (new grass(new_position, world_point));
}

grass::~grass()
{
}